test/compile-test.cc

```diff
@@ -421,3 +421,44 @@ TEST(compile_time_formatting_test, multibyte_fill) {
   EXPECT_EQ("жж42", test_format<8>(FMT_COMPILE("{:ж>4}"), 42));
 }
 #endif
+
+#if FMT_USE_CONSTEXPR_STRING
+
+TEST(compile_test, constexpr_format) {
```

Comment:
I think the long string formatting should be checked, since due to SSO short string tests don't use dynamic allocation in strings.